tap_iterable/__init__.py

```diff
@@ -26,50 +25,13 @@
 
 
 def discover(client):
-    logger.info("Starting discover")
+    LOGGER.info("Starting discover")
     catalog = {"streams": discover_streams(client)}
     json.dump(catalog, sys.stdout, indent=2)
-    logger.info("Finished discover")
+    LOGGER.info("Finished discover")
 
 
-def stream_is_selected(mdata):
-    return mdata.get((), {}).get('selected', False)
-
-
-def get_selected_streams(catalog):
-    selected_stream_names = []
-    for stream in catalog.streams:
-        mdata = metadata.to_map(stream.metadata)
-        if stream_is_selected(mdata):
-            selected_stream_names.append(stream.tap_stream_id)
-    return selected_stream_names
-
-
-def sync(client, catalog, state):
-    selected_stream_names = get_selected_streams(catalog)
-
-    for stream in catalog.streams:
-        stream_name = stream.tap_stream_id
-
-        mdata = metadata.to_map(stream.metadata)
-
-        if stream_name not in selected_stream_names:
-            logger.info("%s: Skipping - not selected", stream_name)
-            continue
-
-        key_properties = metadata.get(mdata, (), 'table-key-properties')
-        singer.write_schema(stream_name, stream.schema.to_dict(), key_properties)
-        logger.info("%s: Starting sync", stream_name)
-        instance = STREAMS[stream_name](client)
-        instance.stream = stream
-        counter_value = sync_stream(state, instance)
-        logger.info("%s: Completed sync (%s rows)", stream_name, counter_value)
-
-    singer.write_state(state)
-    logger.info("Finished sync")
-
-
-@singer.utils.handle_top_exception(logger)
+@singer.utils.handle_top_exception(LOGGER)
 def main():
```

Comment:
```suggestion
@singer.utils.handle_top_exception(logger)
def main():
```
Do u think to have this decorator back in place?

Reply:
Please ignore the above comment. I see the decorator is already in place, couldn't see it in github.dev